x-pack/plugins/beats/server/routes/api/register_enroll_beat_route.js

```diff
@@ -90,7 +86,12 @@ export function registerEnrollBeatRoute(server) {
 
         accessToken = uuid.v4().replace(/-/g, "");
         const remoteAddress = request.info.remoteAddress;
-        await persistBeat(callWithInternalUser, request.payload, request.params.beatId, accessToken, remoteAddress);
+        await persistBeat(callWithInternalUser, {
+          ...omit(request.payload, 'enrollment_token'),
+          id: beatId,
+          access_token: accessToken,
+          host_ip: remoteAddress
+        });
```

Comment:
@exekias In this PR I'm implementing the "Update Beat" API (`PUT /api/beats/agent/{beat UUID}`). That API can take any of the following Beat information in its request body:
* `type`: filebeat, metricbeat, etc.
* `version`: beat version
* `host_name`
* `ephemeral_id`
* `local_configuration_yml`
* `metadata`: arbitrary extra metadata, if we want any

Which of these above properties do you expect to send at **enrollment** time as well?

Reply:
The ones from this list we should be sending for sure are type & hostname. I'm divided on version, but probably it makes sense to include it during enrolment too (also here, as it may change)

Reply:
Cool, thanks. I'll add `version` as a required field to the payload for the enrollment API (it already takes `type` and `host_name` as required fields).

Reply:
Added in 5e63721.